qcodes/plots/qcmatplotlib.py

```diff
@@ -552,29 +594,35 @@ def scale_formatter(i, pos, scale):
                     scales = [10**(3*6 - 3*n) for n in range(len(prefixes))]
 
                     if unit in units_to_scale:
+                        if unit == 'Ohm':
+                            unit = 'Ω'
+
                         scale = 1
                         new_unit = unit
-                        for prefix, threshold, trialscale in zip(prefixes,
-                                                                 thresholds,
-                                                                 scales):
-                            if maxval < threshold:
-                                scale = trialscale
-                                new_unit = prefix + unit
-                                break
-                        # special case the largest
-                        if maxval > thresholds[-1]:
-                            scale = scales[-1]
-                            new_unit = prefixes[-1] + unit
-
-                        tx = ticker.FuncFormatter(
-                            partial(scale_formatter, scale=scale))
+                        # Don't scale the units for a log-plot
+                        if not log_scale:
+                            for prefix, threshold, trialscale in zip(prefixes,
+                                                                     thresholds,
+                                                                     scales):
+                                if maxval < threshold:
+                                    scale = trialscale
+                                    new_unit = prefix + unit
+                                    break
+
+                            # special case the largest
+                            if maxval > thresholds[-1]:
+                                scale = scales[-1]
+                                new_unit = prefixes[-1] + unit
+
+                        if log_scale:
+                            tx = ticker.LogFormatterMathtext()
+                        else:
+                            tx = ticker.FuncFormatter(
+                                partial(scale_formatter, scale=scale))
                         new_label = "{} ({})".format(label, new_unit)
                         if axis in ('x', 'y'):
-                            getattr(subplot,
-                                    "{}axis".format(axis)).set_major_formatter(
-                                tx)
-                            getattr(subplot, "set_{}label".format(axis))(
-                                new_label)
+                            getattr(subplot, "{}axis".format(axis)).set_major_formatter(tx)
+                            getattr(subplot, "set_{}label".format(axis))(new_label)
```

Comment:
Since you're changing this, how about using f strings ;-)

Reply:
Done.